project/src/main/webapp/gmail.js

```diff
@@ -12,193 +12,206 @@
+class GmailAPI {
+    constructor() {
+        this.emailObjects = {};
+    }
 
-    // Authorization scopes required by the API; multiple scopes can be
-    // included, separated by spaces.
-    var SCOPES = 'https://www.googleapis.com/auth/gmail.readonly';
+    getQuery() {
+        this.question = document.getElementById("text-input").value;
+    }
 
-    var authorizeButton = document.getElementById('authorize_button');
-    var signoutButton = document.getElementById('signout_button');
+    getEmailBody(decodedEmail) {
+        const emailBodyStartIndex = decodedEmail.indexOf("Content-Type: text/plain; charset=\"UTF-8\"");
+        const emailBodyEndIndex = (decodedEmail.substring(emailBodyStartIndex)).indexOf("Content-Type: text/html; charset=\"UTF-8\"")
+        let emailBodyValue = decodedEmail.substring(emailBodyStartIndex, emailBodyStartIndex + emailBodyEndIndex);
+        if (emailBodyValue.indexOf("Forwarded message") != -1) {
+            emailBodyValue = emailBodyValue.substring(0, emailBodyValue.indexOf("Forwarded message"));
+        }
+        const quotedPrintableIndex = emailBodyValue.indexOf("quoted-printable");
+        const quotedPrintableLength = "quoted-printable".length;
+        if (quotedPrintableIndex != -1) {
+            emailBodyValue = emailBodyValue.substring(quotedPrintableIndex + quotedPrintableLength);
+        }
+        const contentTypeIndex = emailBodyValue.indexOf("Content-Type: text/plain; charset=\"UTF-8");
+        const contentTypeLength = "Content-Type: text/plain; charset=\"UTF-8".length;
+        if (contentTypeIndex != -1) {
+            emailBodyValue = emailBodyValue.substring(contentTypeIndex + contentTypeLength);
+        }
+        return emailBodyValue;
+    }
 
-    /**
-      *  On load, called to load the auth2 library and API client library.
-      */
-    function handleClientLoad() {
-    gapi.load('client:auth2', initClient);        
+    getEmailDate(decodedEmail) {
+        const emailDateStartIndex = decodedEmail.indexOf("Date:");
+        const emailDateEndIndex = decodedEmail.substring(emailDateStartIndex).indexOf("Message-ID");
+        return decodedEmail.substring(emailDateStartIndex, emailDateStartIndex + emailDateEndIndex);
     }
 
-    /**
-      *  Initializes the API client library and sets up sign-in state
-      *  listeners.
-      */
-    function initClient() {
-        gapi.client.init({
-            apiKey: API_KEY,
-            clientId: CLIENT_ID,
-            discoveryDocs: DISCOVERY_DOCS,
-            scope: SCOPES
-        }).then(function () {
-            // Listen for sign-in state changes.
-            gapi.auth2.getAuthInstance().isSignedIn.listen(updateSigninStatus);
+    getMessageID(decodedEmail) {
+        const MessageIDStartIndex = decodedEmail.indexOf("Message-ID:") + 12;
+        const MessageIDEndIndex = decodedEmail.substring(MessageIDStartIndex).indexOf("Subject:") + MessageIDStartIndex;
```

Comment:
These variables should use lowerCamelCase: https://google.github.io/styleguide/jsguide.html#naming-local-variable-names

Reply:
done!